contracts/src/l1/TokenBridge.sol

```diff
@@ -197,103 +224,158 @@ contract TokenBridge is
         messagingContract.consumeMessageFromL2(l2TokenBridge, payload);
     }
 
-    function deposit(
-        IStaticATokenLM l1Token,
-        uint256 l2Recipient,
-        uint256 amount
-    )
-        onlyApprovedToken(l1Token)
-        onlyValidL2Address(l2Recipient)
-        external
+    function _dynamicToStaticAmount(
+        uint256 amount,
+        address asset,
+        ILendingPool lendingPool
+    ) internal view returns (uint256) {
+        return amount.rayDiv(lendingPool.getReserveNormalizedIncome(asset));
+    }
+
+    function _staticToDynamicAmount(
+        uint256 amount,
+        address asset,
+        ILendingPool lendingPool
+    ) internal view returns (uint256) {
+        return amount.rayMul(lendingPool.getReserveNormalizedIncome(asset));
+    }
+
+    function getCurrentRewardsIndex(address l1AToken)
+        internal
+        view
+        returns (uint256)
     {
-        uint256 rewardsIndex = l1Token.getCurrentRewardsIndex();
+        (
+            uint256 index,
+            uint256 emissionPerSecond,
+            uint256 lastUpdateTimestamp
+        ) = incentivesController.getAssetData(l1AToken);
+        uint256 distributionEnd = incentivesController.DISTRIBUTION_END();
+        uint256 totalSupply = IScaledBalanceToken(l1AToken).scaledTotalSupply();
+
+        if (
+            emissionPerSecond == 0 ||
+            totalSupply == 0 ||
+            lastUpdateTimestamp == block.timestamp ||
+            lastUpdateTimestamp >= distributionEnd
+        ) {
+            return index;
+        }
 
-        l1Token.safeTransferFrom(msg.sender, address(this), amount);
-        sendMessage(l1Token, msg.sender, l2Recipient, amount, block.number, rewardsIndex);
+        uint256 currentTimestamp = block.timestamp > distributionEnd
+            ? distributionEnd
+            : block.timestamp;
+        uint256 timeDelta = currentTimestamp.sub(lastUpdateTimestamp);
+        return
+            emissionPerSecond
+                .mul(timeDelta)
+                .mul(10**uint256(18))
+                .div(totalSupply)
+                .add(index); // 18- precision, should be loaded
     }
 
-    function depositUnderlying(
-        IStaticATokenLM l1Token,
+    function deposit(
+        address l1AToken,
         uint256 l2Recipient,
         uint256 amount,
-        uint16 refferalCode,
+        uint16 referralCode,
         bool fromAsset
-    )
-        onlyApprovedToken(l1Token)
-        onlyValidL2Address(l2Recipient)
-        external
-    {
-        uint256 rewardsIndex = l1Token.getCurrentRewardsIndex();
+    ) external onlyValidL2Address(l2Recipient) {
```

Comment:
Would it be worthwhile to return the static amount that will be minted for the user?

Reply:
problem is lendingPool does not return this amount :/

Reply:
indeed, and we already compute it